packages/cart/src/services/cart-module.ts

```diff
@@ -190,35 +190,62 @@ export default class CartModuleService<
     return createdCarts
   }
 
+  async update(data: CartTypes.UpdateCartDTO[]): Promise<CartTypes.CartDTO[]>
   async update(
-    data: CartTypes.UpdateCartDTO[],
+    cartId: string,
+    data: CartTypes.UpdateCartDataDTO,
     sharedContext?: Context
-  ): Promise<CartTypes.CartDTO[]>
-
+  ): Promise<CartTypes.CartDTO>
   async update(
-    data: CartTypes.UpdateCartDTO,
+    selector: Partial<CartTypes.CartDTO>,
+    data: CartTypes.UpdateCartDataDTO,
     sharedContext?: Context
-  ): Promise<CartTypes.CartDTO>
+  ): Promise<CartTypes.CartDTO[]>
 
   @InjectManager("baseRepository_")
   async update(
-    data: CartTypes.UpdateCartDTO[] | CartTypes.UpdateCartDTO,
```

Comment:
q: should we use InjectTransactionManager? also, maybe split in that case to update and update_ in order to not mess with populate within a transaction